packages/shell-api/src/database.ts

```diff
@@ -817,13 +817,25 @@ export default class Database extends ShellApiWithMongoClass {
       pipeline.push({ $match: matchingFilters });
     }
 
-    return (await this
-      .getSiblingDB('admin')
-      .aggregate(pipeline, {
-        '$readPreference': {
-          'mode': 'primaryPreferred'
-        }
-      })).toArray();
+    const adminDb = this.getSiblingDB('admin');
+    const aggregateOptions = { $readPreference: { mode: 'primaryPreferred' } };
+    let result;
+
+    try {
+      const cursor = await adminDb.aggregate(pipeline, aggregateOptions);
+      result = await cursor.toArray();
+    } catch (error) {
+      if ((error as any).codeName === 'FailedToParse') {
```

Comment:
```suggestion
      if ((error as any)?.codeName === 'FailedToParse') {
```

(I think we do this elsewhere too, probably doesn’t matter much in practice but I think it’s a good standard to stick to)